### ISSUE TITLE
get_log_with_log_prefixes(log) duplicates traces instead of getting prefixes

### ISSUE THREAD

**Issue by Rizzey** [pm4py/pm4py-source#95]
I found this nice little function in the branch with the prediction functions of remaining time. I don't think it agrees with the prefix matrix as mentioned in #94 .
To give an example of what's happening in pseudocode
```pseudocode
log =  [trace]  # a log is a list of traces
trace = <a,b,c,d,e> # a trace is filled with activities a,b,c,d,e in order
extended_log = get_log_with_log_prefixes(log) # calling the function
# the prefix log would have as many traces as activities in the original log
extended_log *expected* = [trace0,trace1,trace2,trace3,trace4] 
                      = [<a>,<a,b>,<a,b,c>,<a,b,c,d>,<a,b,c,d,e>] 
# in reality however, the extended log simply returns... 
extended_log_actual   = [trace4,trace4,trace4,trace4,trace4]
                      = [<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>]
```

Code:
```python
from pm4py.objects.log.util import xes
from pm4py.objects.log.util.get_prefixes import get_log_with_log_prefixes

log_path = os.path.join("./pm4py-source", "tests", "input_data", "running-example.xes")
log = xes_importer.apply(log_path)

#a little lambda to get the trace in terms of activities
activities_in_trace = lambda trace: [event["Activity"] for event in trace]

extended_log = get_log_with_log_prefixes(log)
#as per the get_prefix_matrix function, i would expect this gets
"""
log =  [trace]
trace0 = <a,b,c,d,e>
extended_log_expected = [trace0,trace1,trace2,trace3,trace4]
                      = [<a>,<a,b>,<a,b,c>,<a,b,c,d>,<a,b,c,d,e>]
extended_log_actual   = [trace0,trace1,trace2,trace3,trace4]
                      = [<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>,<a,b,c,d,e>]
"""
acts = [activities_in_trace(trace) for trace in log]
prefix_acts = [activities_in_trace(trace) for trace in extended_log]
for i in range(8):
    acts[0] == prefix_acts[i] 
```
and this returns 
```
>>> True
True
True
True
True
True
True
True
```
Is this intentional (if it is, apologies for the two misunderstandings in two days) or a bug?

**Comment by s-j-v-zelst** [comment]
Dear @Rizzey,

thank you for signaling this.
This was a bug, in the get_prefix() code.
This was fixed in commit d32613f65af947bb14f794119c27ccf0a59c7af8

**Comment by Javert899** [comment]
Released version 1.1.20 where this has been addressed
